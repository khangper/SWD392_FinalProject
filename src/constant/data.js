export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
