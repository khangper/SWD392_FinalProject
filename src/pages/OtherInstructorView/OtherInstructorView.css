/* .card .other_instructor_view-container {
        max-width: 1200px;
        margin: 0 auto;
        background: #f7f7f7;
        padding-bottom: 20px;
        transition: padding-left 0.4s;
        overflow-y: auto;
        flex: 1 1;

    } */
.other_instructor_view-wrapper {
  position: -webkit-sticky;
  position: sticky;
}

.other_instructor_view-report_profile {
  text-decoration: none;
  font-size: 13px;
}

.report-profile{
    color: #fff !important;
}
.other_instructor_view-subcrise {
  display: flex;
  gap: 20px;
  margin-right: 70px;
  padding-left: 0 !important;
}


.other_instructor_view-report_profile span i {
  color: #fff;
}
.other_instructor_view-row2 {
  display: flex;
  justify-content: space-between;
}

.other_instructor_view-col-md-12 {
  flex: 0 0 100%;
  max-width: 100%;
}

.other_instructor_view-section {
  background-color: #333;
  padding: 20px;
  border-radius: 5px;
  color: #fff;
}

.other_instructor_view-avatar {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
}

.other_instructor_view-img img {
  width: 80px;
  height: 80px;
  border-radius: 50%;
  margin-right: 15px;
}

.other_instructor_view-name-career h2 {
  font-size: 24px;
  margin: 0;
}

.other_instructor_view-name-career span {
  display: block;
  font-size: 16px;
  color: #bbb;
}

.other_instructor_view-more_info {
  list-style: none;
  padding: 0;
  margin: 0;
  display: flex;
  justify-content: space-between;
}

.other_instructor_view-more_info li {
  border: 1px solid #bbb;
  width: 100px;
}

.other_instructor_view-more_info li:last-child {
  margin-right: 0; /* Remove right margin from the last item */
}

.other_instructor_view-enroll_label_more_info {
  text-align: center;
}

.other_instructor_view-enroll_label_more_info {
  text-align: center;
}

.other_instructor_view-label {
  font-size: 14px;
  color: #bbb;
}

.other_instructor_view-number {
  font-size: 18px;
  margin-top: 5px;
}

.btn500 {
  display: inline-block;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  background: #f00;
  color: #fff;
  cursor: pointer;
  transition: background 0.3s;
  width: 120px;
}


.btn500:hover {
  background: #d00;
}

.msg125 {
  background: #333;
  border: 1px solid #fff;
}

.msg125:hover {
  background: #444;
}

.other_instructor_view-col-lg-6 {
  order: 1; /* Ensure this column appears first on the left */
  /* padding-left: 220px; */
}

.other_instructor_view-col-lg-5 {
  order: 2; /* Ensure this column appears second on the right */
  display: flex;
  flex-direction: column;
  gap: 30px;
  align-items: flex-end;
}

.social-icons .oiv-fb {
  font-size: 20px;
  float: none;
  width: 35px;
  height: 36px;
  border-radius: 5px;
  text-align: center;
  /* line-height: 40px; */
  padding: 9px 0;
  color: #fff;
  transition: all 0.3s ease-in-out;
  background: #3b5998;
}
.social-icons .oiv-x {
  font-size: 20px;
  float: none;
  width: 35px;
  height: 36px;
  border-radius: 5px;
  text-align: center;
  /* line-height: 40px; */
  padding: 9px 0;
  color: #fff;
  transition: all 0.3s ease-in-out;
  background: #1da1f2;
  /* background: #262424;
    border: 1px solid #262424; */
}

.social-icons .oiv-ln {
  font-size: 20px;
  float: none;
  width: 35px;
  height: 36px;
  border-radius: 5px;
  text-align: center;
  /* line-height: 40px; */
  padding: 9px 0;
  color: #fff;
  transition: all 0.3s ease-in-out;
  background: #8d6cab;
}

.social-icons .oiv-yt {
  font-size: 20px;
  float: none;
  width: 35px;
  height: 36px;
  border-radius: 5px;
  text-align: center;
  /* line-height: 40px; */
  padding: 9px 0;
  color: #fff;
  transition: all 0.3s ease-in-out;
  background: #ff0000;
}

.social-icons{
    margin-right: 0 !important;
    display: flex;
    flex-direction: row;
    gap: 10px !important;
    margin-right: 70px !important;
}

/* NAV */
.other_instructor_view-nav {
  display: flex;
  justify-content: flex-start;
  padding: 10px 20px;
  border-bottom: 1px solid #ddd;
}

.tab {
  padding: 10px 20px;
  cursor: pointer;
  position: relative;
  margin-right: 20px; /* Add space between the tabs */
}

.tab div {
  color: #333;
  font-size: 16px;
}

.tab.active div {
  font-weight: bold;
}

.tab.active::after {
  content: "";
  position: absolute;
  bottom: -1px;
  left: 0;
  width: 100%;
  height: 2px;
  background: red;
}

.about-tab {
  padding: 20px;
}

.other_instructor_view-about_me {
  margin: 20px 0;
}

.label-about h3 {
  font-size: 24px;
  margin-bottom: 15px;
  color: #333;
}

.label-about p {
  font-size: 16px;
  line-height: 1.6;
  color: #666;
}

/* COURSE */

.tab-pane .other_instructor_view-crse_content {
  padding: 20px;
}

.other_instructor_view-crse_content h3 {
  font-size: 24px;
  margin-bottom: 20px;
}

.other_instructor_view-list_courses {
  display: flex;
  flex-wrap: wrap;
  gap: 20px;
}

.other_instructor_view-row {
  display: flex;
  flex-wrap: wrap;
  gap: 20px;
  justify-content: space-between;
}

.oiv-course {
  background: #fff;
  border: 1px solid #e5e5e5;
  border-radius: 8px;
  overflow: hidden;
  width: calc(25% - 20px);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.oiv-course-detail {
  display: flex;
  flex-direction: column;
}

.fcrse_img {
  position: relative;
  display: block;
  overflow: hidden;
}

.fcrse_img img {
  width: 100%;
  height: auto;
  display: block;
}

.oiv-course-overlay {
  position: absolute;
  top: 10px;
  padding: 10px;
  left: 10px;
  right: 15px;
  bottom: 180px;
}

.oiv-badge_seller {
  border-radius: 3px;
  font-size: 10px;
  position: absolute;
  padding: 2px 10px;
  background: #fa8305;
  color: #fff;
  font-family: "Roboto", sans-serif;
  font-weight: 600;
  right: 0;
  top: 10px;
  text-transform: uppercase;
  box-shadow: 0 0 1px 1px rgba(20, 23, 28, 0.1),
    0 3px 1px 0 rgba(20, 23, 28, 0.1);
}

.crse_reviews {
  color: #fff;
  background: #fdcc0d;
  display: inline-flex;
  padding: 3px 15px 3px 10px;
  border-radius: 3px;
  font-size: 14px;
  font-weight: 500;
  font-family: "Roboto", sans-serif;
}

.crse_reviews .starIcon {
  width: 15px;
  height: 15px;
  top: 1px;
  position: relative;
  margin-right: 5px;
}

.crse_timer {
  bottom: -120px;
  position: absolute;
  margin-bottom: 10px;
  right: 0;
  margin-right: 10px;
  color: #fff;
  background: rgba(51, 51, 51, 0.8);
  display: inline-block;
  padding: 3px 10px;
  border-radius: 3px;
  font-size: 12px;
  font-weight: 500;
  font-family: "Roboto", sans-serif;
}
.other_instructor_view-time_view {
  margin-bottom: 10px;
  font-size: 12px;
  margin-top: 5px;
  font-weight: 400;
  font-family: "Roboto", sans-serif;
  color: #686f7a;
  text-align: left;
  display: block;
}

.fcrse_contents {
  padding: 5px 15px;
  float: left;
  width: 100%;
  position: relative;
}

.fcrse_content {
  padding: 10px;
  display: flex;
  justify-content: space-between;
}

.eps_dots {
  position: relative;
  float: right;
  display: inline-block;
}

.eps_dots .oiv-dropdown-button {
  font-size: 20px;
  color: #afafaf !important;
  text-decoration: none;
}

.oiv-dropdown-content {
  display: none;
  position: absolute;
  background-color: #fff !important;
  min-width: 160px;
  box-shadow: 0 2px 3px 0 rgba(34, 36, 38, 0.15);
  border: 1px solid rgba(34, 36, 38, 0.15);
  z-index: 3;
  right: 0;
  border-radius: 3px;
}

.oiv-dropdown-content span {
  color: #686f7a;
  font-size: 14px;
  font-weight: 400;
  font-family: "Roboto", sans-serif;
  padding: 3px;
  text-decoration: none;
  display: block;
  cursor: pointer;
}

.oiv-dropdown-content span img {
  width: 14px;
  height: 14px;
  margin-right: 10px;
}

.eps_dots:hover .oiv-dropdown-content {
  display: block;
}

.dropdown-content span {
  display: block;
  padding: 10px;
  font-size: 14px;
  color: #333;
  cursor: pointer;
}

.dropdown-content span:hover {
  background: #f5f5f5;
}

.other_instructor_view-time_view {
  font-size: 12px;
  color: #999;
  margin-bottom: 5px;
}

.oiv-detail {
  display: block;
  font-size: 16px;
  font-weight: 600;
  font-family: "Roboto", sans-serif;
  color: #333;
  text-align: left;
  margin-bottom: 7px;
  line-height: 24px;
}

.oiv-career {
  display: block;
  font-size: 13px;
  font-weight: 400;
  font-family: "Roboto", sans-serif;
  color: #686f7a;
  text-align: left;
  padding-top: 2px;
}

.other_instructor_view-publisher {
  float: left;
  width: 100%;
  margin-top: 15px;
}

.oiv-publisher {
  margin-bottom: 0;
  /* margin-top: 18px; */
  font-size: 14px;
  font-family: "Roboto", sans-serif;
  float: left;
  color: #686f7a;
  text-align: left;
  line-height: 24px;
}

.oiv-publisher a {
  color: #333;
  font-weight: 500;
}

.oiv-price {
  float: right;
  font-size: 18px;
  font-weight: 500;
  font-family: "Roboto", sans-serif;
  color: #333;
  text-align: right;
}

.oiv-cart {
  display: none;
  border: none;
  padding-left: 100px;
  background: white;
}

.oiv-course-detail:hover .oiv-cart {
  display: block;
}

.oiv-cart-icon {
  width: 17px;
}

/* DISCUSSION */

.tab-pane .fade {
  padding: 20px;
}

.student_reviews {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  padding: 20px;
  margin-bottom: 20px;
}

.other_instructor_view-row-discussion {
  display: flex;
  flex-direction: column;
}

.review_right_heading h3 {
  margin-bottom: 20px;
  font-size: 24px;
  font-weight: 600;
}

.oiv-comment {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  background-color: #fff;
  padding: 15px;
}

.cmnt_group {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.oiv-img160 img {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  margin-right: 10px;
}

.oiv-cmnt-box {
  width: 100%;
  padding: 10px;
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  font-size: 14px;
  outline: none;
  resize: none;
  background: #f7f7f7;
}

.oiv-cmnt-btn {
  align-self: flex-end;
  padding: 8px 16px;
  background-color: #ff3b30;
  color: #fff;
  border: none;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
}

.oiv-history-comment {
  margin-top: 20px;
}

.oiv-review_item {
  padding: 15px;
  background-color: #fff;
}
.oiv-review_item:first-child{
  border-bottom: 1px solid #efefef;
}
.oiv-review_reply {
  padding-left: 50px;
  background-color: #fff;
}

.oiv-review_usr_dt {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.oiv-review_usr_dt img {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  margin-right: 10px;
}

.oiv-cmnt_name h4 {
  margin: 0;
  font-size: 16px;
  font-weight: 600;
}

.oiv-cmnt_time {
  font-size: 12px;
  color: #888;
}

.eps_dots {
  margin-left: auto;
  position: relative;
}

.eps_dots a {
  color: #888;
  font-size: 20px;
}

.eps_dots:hover .oiv-dropdown-content {
  display: block;
}

.oiv-dropdown-content span {
  display: block;
  padding: 10px;
  cursor: pointer;
}

.oiv-dropdown-content span:hover {
  background-color: #f9f9f9;
}

.oiv-cmnt_text_message {
  font-size: 14px;
  margin-bottom: 10px;
}

.oiv-contact_icon {
  display: flex;
  align-items: center;
}

.oiv-contact_icon img {
  width: 20px;
  margin-right: 10px;
}

.oiv-report {
  display: flex;
  align-items: center;
  color: #888;
  font-size: 14px;
  margin-right: 15px;
}

.oiv-report i {
  margin-right: 5px;
}

.oiv-report.ml-3 {
  margin-left: 20px;
}
