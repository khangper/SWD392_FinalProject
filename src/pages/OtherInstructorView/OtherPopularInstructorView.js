import React, { useEffect, useState } from "react";
import "./OtherInstructorView.css";
import facebook from "../../assets/facebook.png";
import x from "../../assets/x.png";
import linkedin from "../../assets/linkedin.png";
import youtube from "../../assets/youtube.png";
import ratingStar from "../../assets/rating.png";
import share from "../../assets/share.png";
import saved_course from "../../assets/saved-course.png";
import not_interested from "../../assets/not-interested.png";
import report from "../../assets/report.png";
import card_icon from "..//../assets/cart-icon.png";
import like from "../../assets/like.png";
import dislike from "../../assets/dislike.png";
import heart from "../../assets/heart.png";
import edit from "../../assets/edit.png";
import deletee from "../../assets/delete.png";
import { Link, useParams } from "react-router-dom";
import { PATH_NAME } from "../../constant/pathname";
import { useDispatch, useSelector } from "react-redux";
import { fetchInstructorByIdRequest } from "../../redux/reduxActions/instructorActions";
import { fetchCoursesDRequest } from "../../redux/reduxActions/CoursesDAction";
import { fetchCommentsRequest } from "../../redux/reduxActions/CommentAction";
import { fetchPopularInstructorDetailRequest } from "../../redux/reduxActions/homeActions/HomePopularInstructorAction";
const OtherPopularInstructorView = () => {
    const dispatch = useDispatch();
    const { id } = useParams();
    const { popularInstructors } = useSelector(
        (state) => state.home_popularinstructor
    );
    const popularInstructorDetail = popularInstructors.find(popularInstructor => popularInstructor.id === id);
    const [oivtab, setOivtab] = useState("about");
    useEffect(() => {
        if (!popularInstructorDetail) {
            dispatch(fetchPopularInstructorDetailRequest(id));
        }
    }, [dispatch, id, popularInstructorDetail]);
    const { coursesd } = useSelector((state) => state.coursesd);

    useEffect(() => {
        dispatch(fetchCoursesDRequest());
    }, [dispatch]);

    const { comments } = useSelector((state) => state.comments);

    useEffect(() => {
        dispatch(fetchCommentsRequest());
    }, [dispatch]);

    const handleoivTabChange = (tab) => {
        setOivtab(tab);
    };

    return (
        <div>
            <div className="other_instructor_view-wrapper">
                <div className="card">
                    <div className="other_instructor_view-container">
                        <div className="other_instructor_view-row">
                            <div className="other_instructor_view-col-md-12">
                                <div className="other_instructor_view-section">
                                    <div className="other_instructor_view-row2">
                                        <div className="other_instructor_view-col-lg-6">
                                            <div className="other_instructor_view-avatar">
                                                <div className="other_instructor_view-img">
                                                    <img src={popularInstructorDetail.imgSrc} alt={popularInstructorDetail.name} />
                                                </div>
                                                <div className="other_instructor_view-name-career">
                                                    <h2>{popularInstructorDetail.name}</h2>
                                                    <span>{popularInstructorDetail.title}</span>
                                                </div>
                                            </div>
                                            <ul className="other_instructor_view-more_info">
                                                <li>
                                                    <div className="other_instructor_view-enroll_label_more_info">
                                                        <div className="other_instructor_view-label">
                                                            Enroll Students
                                                        </div>
                                                        <div className="other_instructor_view-number">
                                                            200k
                                                        </div>
                                                    </div>
                                                </li>
                                                <li>
                                                    <div className="other_instructor_view-enroll_label_more_info">
                                                        <div className="other_instructor_view-label">
                                                            Courses
                                                        </div>
                                                        <div className="other_instructor_view-number">
                                                            20
                                                        </div>
                                                    </div>
                                                </li>
                                                <li>
                                                    <div className="other_instructor_view-enroll_label_more_info">
                                                        <div className="other_instructor_view-label">
                                                            Reviews
                                                        </div>
                                                        <div className="other_instructor_view-number">
                                                            115K
                                                        </div>
                                                    </div>
                                                </li>
                                                <li>
                                                    <div className="other_instructor_view-enroll_label_more_info">
                                                        <div className="other_instructor_view-label">
                                                            Subscribers
                                                        </div>
                                                        <div className="other_instructor_view-number">
                                                            452K
                                                        </div>
                                                    </div>
                                                </li>
                                            </ul>
                                        </div>
                                        <div className="other_instructor_view-col-lg-5">
                                            <a
                                                href="#"
                                                className="other_instructor_view-report_profile"
                                            >
                                                <span>
                                                    <i className="uil uil-windsock"></i>
                                                </span>
                                                <Link to={PATH_NAME.REPORT_HISTORY} className="report-profile">
                                                    Report Profile
                                                </Link>
                                            </a>
                                            <div className="other_instructor_view-social">
                                                <ul className="social-icons">
                                                    <a href="#" className="oiv-fb">
                                                        <img src={facebook} />
                                                    </a>
                                                    <a href="#" className="oiv-x">
                                                        <img src={x} />
                                                    </a>
                                                    <a href="#" className="oiv-ln">
                                                        <img src={linkedin} />
                                                    </a>
                                                    <a href="#" className="oiv-yt">
                                                        <img src={youtube} />
                                                    </a>
                                                </ul>
                                            </div>
                                            <ul className="other_instructor_view-subcrise">
                                                <button className="subscribe-btn btn500">
                                                    Subscribe
                                                </button>
                                                <Link to={PATH_NAME.MESSAGE}>
                                                    <button className="msg125 btn500">Message</button>
                                                </Link>
                                            </ul>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="other_instructor_view-navigation">
                    <div className="other_instructor_view-container">
                        <div className="other_instructor_view-row">
                            <div className="other_instructor_view-col-lg-12">
                                <div className="other_instructor_view-course_tabs">
                                    <nav>
                                        <div className="other_instructor_view-nav">
                                            <div
                                                className={`tab ${oivtab === "about" ? "active" : ""}`}
                                                onClick={() => handleoivTabChange("about")}
                                            >
                                                <div className="oiv-nav-about">About</div>
                                            </div>
                                            <div
                                                className={`tab ${oivtab === "courses" ? "active" : ""
                                                    }`}
                                                onClick={() => handleoivTabChange("courses")}
                                            >
                                                <div className="oiv-nav-courses">Courses</div>
                                            </div>
                                            <div
                                                className={`tab ${oivtab === "discussion" ? "active" : ""
                                                    }`}
                                                onClick={() => handleoivTabChange("discussion")}
                                            >
                                                <div className="oiv-nav-discussion">Discussion</div>
                                            </div>
                                        </div>
                                        {oivtab === "about" && (
                                            <div className="about-tab">
                                                <div className="tab-pane fade show active">
                                                    <div className="other_instructor_view-about_me">
                                                        <div className="label-about">
                                                            <h3>About Me</h3>
                                                            <p>
                                                                Lorem ipsum dolor sit amet, consectetur
                                                                adipiscing elit. Vestibulum scelerisque nibh sed
                                                                ligula blandit, quis faucibus lorem
                                                                pellentesque. Suspendisse pulvinar dictum
                                                                pellentesque. Vestibulum at sagittis lectus, sit
                                                                amet aliquam turpis. In quis elit tempus, semper
                                                                justo vitae, lacinia massa. Etiam sagittis quam
                                                                quis fermentum lacinia. Curabitur blandit sapien
                                                                et risus congue viverra. Mauris auctor risus sit
                                                                amet cursus sollicitudin. Lorem ipsum dolor sit
                                                                amet, consectetur adipiscing elit. Nulla feugiat
                                                                sodales massa, in viverra dolor condimentum ut.
                                                                In imperdiet, justo nec volutpat blandit, tellus
                                                                justo tempor quam, sed pretium nibh nunc nec
                                                                mauris. Mauris vel malesuada magna. Quisque
                                                                iaculis molestie purus, non luctus mauris porta
                                                                id. Maecenas imperdiet tincidunt mauris
                                                                vestibulum vulputate. Aenean sollicitudin
                                                                pretium nibh, et sagittis risus tincidunt ac.
                                                                Phasellus scelerisque rhoncus massa, ac euismod
                                                                massa pharetra non. Phasellus dignissim, urna in
                                                                iaculis varius, turpis libero mollis velit, sit
                                                                amet euismod arcu mi ac nibh. Praesent tincidunt
                                                                eros at ligula pellentesque elementum. Fusce
                                                                condimentum enim a tellus egestas, sit amet
                                                                rutrum elit gravida. Pellentesque in porta
                                                                sapien. Fusce tristique maximus ipsum et mollis.
                                                                Sed at massa ac est dapibus vulputate at eu
                                                                nibh.
                                                            </p>
                                                        </div>
                                                    </div>
                                                </div>
                                            </div>
                                        )}
                                        {oivtab === "courses" && (
                                            <div className="tab-pane fade">
                                                <div className="other_instructor_view-crse_content">
                                                    <h3>My courses (8)</h3>
                                                    <div className="other_instructor_view-list_courses">
                                                        <div className="other_instructor_view-row">
                                                            {coursesd.map((course) => (
                                                                <div className="oiv-course" key={course.id}>
                                                                    <div className="oiv-course-detail">
                                                                        <a href="course_detail_view.html" className="fcrse_img">
                                                                            <img src={course.image} alt='' />
                                                                            <div className="oiv-course-overlay">
                                                                                {course.bestseller && (
                                                                                    <div className="oiv-badge_seller">Bestseller</div>
                                                                                )}
                                                                                <div className="crse_reviews">
                                                                                    <img className="starIcon" src={ratingStar} alt="rating" />
                                                                                    {course.ratingStar}
                                                                                </div>
                                                                                <div className="crse_timer">{course.time}</div>
                                                                            </div>
                                                                        </a>
                                                                        <div className="fcrse_contents">
                                                                            <div className="fcrse_content">
                                                                                <div className="other_instructor_view-time_view">
                                                                                    <span className="view">{course.view} • </span>
                                                                                    <span className="time">{course.day}</span>
                                                                                </div>
                                                                                <div className="eps_dots more_dropdown">
                                                                                    <a href="#" className="oiv-dropdown-button">⋮</a>
                                                                                    <div className="oiv-dropdown-content">
                                                                                        <span><img src={share} alt="share" />Share</span>
                                                                                        <span><img src={saved_course} alt="save" />Save</span>
                                                                                        <span><img src={not_interested} alt="not interested" />Not Interested</span>
                                                                                        <span><img src={report} alt="report" />Report</span>
                                                                                    </div>
                                                                                </div>
                                                                            </div>
                                                                            <a href="course_detail_view.html" className="oiv-detail">
                                                                                {course.detail}
                                                                            </a>
                                                                            <a href="#" className="oiv-career">
                                                                                {course.career}
                                                                            </a>
                                                                            <div className="other_instructor_view-publisher">
                                                                                <p className="oiv-publisher">
                                                                                    By <a href="#">{course.author}</a>
                                                                                </p>
                                                                                <div className="oiv-price">{course.price}</div>
                                                                                <button className="oiv-cart" title="cart">
                                                                                    <img src={card_icon} alt="cart" className="oiv-cart-icon" />
                                                                                </button>
                                                                            </div>
                                                                        </div>
                                                                    </div>
                                                                </div>
                                                            ))}

                                                        </div>
                                                    </div>
                                                </div>
                                            </div>
                                        )}
                                        {oivtab === "discussion" && (
                                            <div className="tab-pane fade">
                                                <div className="student_reviews">
                                                    <div className="other_instructor_view-row-discussion">
                                                        <div className="other_instructor_view-col-lg-12">
                                                            <div className="review_right">
                                                                <div className="review_right_heading">
                                                                    <h3>Discussions</h3>
                                                                </div>
                                                            </div>
                                                            <div className="oiv-comment">
                                                                <div className="cmnt_group">
                                                                    <div className="oiv-img160">
                                                                        <img
                                                                            src="https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-1.jpg"
                                                                            alt=""
                                                                        />
                                                                    </div>
                                                                    <textarea
                                                                        className="oiv-cmnt-box"
                                                                        placeholder="Add a public comment"
                                                                    ></textarea>
                                                                </div>
                                                                <button className="oiv-cmnt-btn" type="submit">
                                                                    Comment
                                                                </button>
                                                            </div>
                                                            {comments.map((comment) => (
                                                                <div className="oiv-history-comment" key={comment.id}>
                                                                    <div className="oiv-review_item">
                                                                        <div className="oiv-review_usr_dt">
                                                                            <img src={comment.userImage} alt="" />
                                                                            <div className="oiv-cmnt_name">
                                                                                <h4 className="oiv-tutor_name">{comment.userName}</h4>
                                                                                <span className="oiv-cmnt_time">{comment.timeAgo}</span>
                                                                            </div>
                                                                            <div className="eps_dots more_dropdown">
                                                                                <a href="#" className="oiv-dropdown-button">⋮</a>
                                                                                <div className="oiv-dropdown-content">
                                                                                    <span><img src={edit} alt="edit" /> Edit</span>
                                                                                    <span><img src={deletee} alt="delete" /> Delete</span>
                                                                                </div>
                                                                            </div>
                                                                        </div>
                                                                        <p className="oiv-cmnt_text_message">{comment.message}</p>
                                                                        <div className="oiv-contact_icon">
                                                                            <a href="#" className="oiv-report"><img src={like} alt="like" /> {comment.likes}</a>
                                                                            <a href="#" className="oiv-report"><img src={dislike} alt="dislike" /> {comment.dislikes}</a>
                                                                            <a href="#" className="oiv-report"><img src={heart} alt="heart" /></a>
                                                                            <a href="#" className="oiv-report ml-3">Reply</a>
                                                                        </div>
                                                                    </div>
                                                                    {comment.replies && comment.replies.map((reply) => (
                                                                        <div className="oiv-review_reply" key={reply.id}>
                                                                            <div className="oiv-review_item">
                                                                                <div className="oiv-review_usr_dt">
                                                                                    <img src={reply.userImage} alt="" />
                                                                                    <div className="oiv-cmnt_name">
                                                                                        <h4 className="oiv-tutor_name">{reply.userName}</h4>
                                                                                        <span className="oiv-cmnt_time">{reply.timeAgo}</span>
                                                                                    </div>
                                                                                    <div className="eps_dots more_dropdown">
                                                                                        <a href="#" className="oiv-dropdown-button">⋮</a>
                                                                                        <div className="oiv-dropdown-content">
                                                                                            <span><img src={deletee} alt="delete" /> Delete</span>
                                                                                        </div>
                                                                                    </div>
                                                                                </div>
                                                                                <p className="oiv-cmnt_text_message">{reply.message}</p>
                                                                                <div className="oiv-contact_icon">
                                                                                    <a href="#" className="oiv-report"><img src={like} alt="like" /> {reply.likes}</a>
                                                                                    <a href="#" className="oiv-report"><img src={dislike} alt="dislike" /> {reply.dislikes}</a>
                                                                                    <a href="#" className="oiv-report"><img src={heart} alt="heart" /></a>
                                                                                    <a href="#" className="oiv-report ml-3">Reply</a>
                                                                                </div>
                                                                            </div>
                                                                        </div>
                                                                    ))}
                                                                </div>
                                                            ))}
                                                        </div>
                                                    </div>
                                                </div>
                                            </div>
                                        )}
                                    </nav>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default OtherPopularInstructorView;
