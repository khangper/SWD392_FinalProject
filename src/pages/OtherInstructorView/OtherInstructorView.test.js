import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import OtherInstructorView from "./OtherInstructorView";
import { screen } from "@testing-library/react";

describe("OtherInstructorView", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the OtherInstructorView component", async () => {
        const { container } = renderWithProviders(<OtherInstructorView />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})