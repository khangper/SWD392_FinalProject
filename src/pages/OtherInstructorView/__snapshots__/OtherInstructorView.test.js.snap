// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherInstructorView renders the OtherInstructorView component 1`] = `<div />`;
